webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +96,776 @@ public WebSiteSessionController createComponentSessionController(
-          String pubId = null;
-          Collection<NodePK> listNodePk;
-          while (end != -1) {
-            pubId = listeSite.substring(begin, end); // pubId
+    String pubId = scc.createWebSite(site);
+    if (!listIcons.isEmpty()) {
+      scc.associateIcons(site.getId(), listIcons);
+    }
 
-            scc.removePublicationToFolder(pubId, id);
+              /* publications : classer le site dans les themes cochés */
+    ArrayList<String> arrayToClassify = new ArrayList<>();
+    boolean publish = false;
+    for (String idTopic : listOfTopics.split(",")) {
+      if (!idTopic.isEmpty()) {
+        scc.addPublicationToFolder(pubId, idTopic);
+        publish = true;
+      }
+    }
 
-            /* isPublished dans un autre theme */
-            listNodePk = scc.getAllFatherPK(pubId);
-            if (listNodePk.size() == 0) {
-              PublicationDetail pubDetail = scc.getPublicationDetail(pubId);
-              idSiteToDeClassify = pubDetail.getVersion();
-              arrayToDeClassify.add(idSiteToDeClassify);
-            }
-
-            begin = end + 1;
-            end = listeSite.indexOf(',', begin);
-          }
+    if (publish) {
+      arrayToClassify.add(site.getId());
+      scc.publish(arrayToClassify);
+    }
+  }
 
-          if (arrayToDeClassify.size() > 0) {
-            scc.dePublish(arrayToDeClassify); // set etat du site a 0
-          }
-          action = "Search";
-        }
+  private String uploadFile(final WebSiteSessionController scc, final HttpRequest request)
+      throws Exception {
 
-        if (id == null) {
-          id = "0";
-        }
+    List<FileItem> items = request.getFileItems();
+    boolean status = true;
+    String thePath = FileUploadUtil.getParameter(items, "path");
+    scc.checkPath(thePath);
+    FileItem item = FileUploadUtil.getFile(items);
+    if (item != null) {
+      status = (scc.addFileIntoWebSite(thePath, item) == 0);
+    }
 
-        FolderDetail webSitesCurrentFolder = scc.getFolder(id);
-        scc.setSessionTopic(webSitesCurrentFolder);
-        request.setAttribute("CurrentFolder", webSitesCurrentFolder);
+    request.setAttribute("UploadOk", status);
 
-        destination = "/webSites/jsp/organize.jsp?Action=" + action + "&Id=" + id + "&Path=" + path;
-      } else if ("AddTopic".equals(function)) {
+    return  "/webSites/jsp/uploadFile.jsp?path=" + thePath;
+  }
 
-        String action = request.getParameter("Action");// =Add
-        String fatherId = request.getParameter("Id");
-        String newTopicName = request.getParameter("Name");
-        String newTopicDescription = request.getParameter("Description");
+  private String manageSite(final WebSiteSessionController scc, final HttpRequest request)
+      throws WebSitesException {
+    String action = request.getParameter("Action");
+    if ("addBookmark".equals(action)) {
+      addBookmarkToSite(scc, request);
+    } else if (action.equals("deleteWebSites")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "deleteWebSites" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1132)